packages/angular_devkit/build_angular/src/webpack/configs/styles.ts

```diff
@@ -148,6 +148,12 @@ export function getStylesConfig(wco: WebpackConfigOptions): Configuration {
     !buildOptions.sourceMap.hidden
   );
 
+  if (!componentsSourceMap && cssSourceMap) {
+    logger.warn(
+      'Components styles sourcemaps are not generated when styles optimization is enabled.',
```

Comment:
Based on the logic for `componentsSourceMap`, enabling hidden source maps will also trigger this behavior. Maybe add that to the warning message also?